core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/SplitAssigner.java

```diff
@@ -69,19 +69,27 @@ record AssignmentResult(
             List<PartitionUpdate> partitionUpdates,
             ImmutableIntArray sealedPartitions)
     {
+        public static final AssignmentResult EMPTY_ASSIGNMENT_RESULT = new AssignmentResult(ImmutableList.of(), false, ImmutableList.of(), ImmutableIntArray.of());
+
         public AssignmentResult
         {
             partitionsAdded = ImmutableList.copyOf(requireNonNull(partitionsAdded, "partitionsAdded is null"));
             partitionUpdates = ImmutableList.copyOf(requireNonNull(partitionUpdates, "partitionUpdates is null"));
         }
 
+        boolean isEmpty()
+        {
+            return this == EMPTY_ASSIGNMENT_RESULT || (partitionsAdded.isEmpty() && !noMorePartitions && partitionUpdates.isEmpty() && sealedPartitions.isEmpty());
```

Comment:
is `this == EMPTY_ASSIGNMENT_RESULT ||` optimization important?

Reply:
Probably not too much - but it is not super harmful I think - I would leave it unless you strongly disagree.

Reply:
it feels unnecessary complexity

of you remove EMPTY_ASSIGNMENT_RESULT constant, you don't need `empty` tracking in the builder
and `isEmpty` becomes simple. sounds like a win

Reply:
I will drop